iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java

```diff
@@ -237,13 +239,51 @@ void updateUser(EventProcessingRequest request) throws IOException {
             addUserIdentitiesToRequest(userUpdateRequest, request);
             if (!isEmpty(userUpdateRequest.email) || !isEmpty(userUpdateRequest.userId)) {
                 userUpdateRequest.dataFields = convertAttributes(request.getUserAttributes(), shouldCoerceStrings(request));
+                if (request.getUserAttributeLists() != null && !request.getUserAttributeLists().isEmpty()) {
+                    Map<String, List<Object>> processedUserAttributeList = processUserAttributesList(request.getUserAttributeLists(), shouldCoerceStrings(request));
+                    for (Map.Entry<String, List<Object>> entry : processedUserAttributeList.entrySet()) {
+                        String key = entry.getKey();
+                        Object value = entry.getValue();
+                        userUpdateRequest.dataFields.put(key, value);
+                    }
+                }
                 Call<IterableApiResponse> preparedCall = iterableService.userUpdate(getApiKey(request), userUpdateRequest);
                 Response<IterableApiResponse> response = makeIterableRequest(preparedCall, request.getId());
                 handleIterableResponse(preparedCall, response, request.getId());
             }
         }
     }
 
+    /**
+     * Returns a type-corrected userAttributeList
+     *
+     * @param userAttributesList Map of <keys>: List<String values>
+     * @param shouldCoerceStrings boolean setting in mParticle
+     * @return a type-corrected map
+     */
+    private Map<String, List<Object>> processUserAttributesList(Map<String, List<String>> userAttributesList, boolean shouldCoerceStrings) {
+        Map<String, List<Object>> convertedAttributeList = new HashMap<>();
+        userAttributesList.forEach((key, value)-> {
+            value.forEach((attribute)-> {
+                if (!convertedAttributeList.containsKey(key)) {
+                    convertedAttributeList.put(key, new ArrayList<>());
+                }
+                if (isValidJson(attribute)) {
+                    Map<String, Object> typeCorrectedMap = jsonToMap(attribute);
+                    convertedAttributeList.get(key).add(typeCorrectedMap);
+                } else {
+                    // Retrofit to use convertAttributes
+                    Map<String, String> temp = new HashMap<>();
+                    temp.put(key, attribute);
+                    Map<String, Object> converted = convertAttributes(temp, shouldCoerceStrings);
```

Comment:
What does `convertAttributes` do? Does it convert the attribute to a `Map` like the code in the `if` block?

Reply:
All data inbound from mParticle comes in as a string, so `convertAttributes` is responsible for coercing those string values into the intended types. 

`convertAttributes` takes a `Map` and ensures:
1) the key is not a reserved key in mParticle (i.e. `phoneNumber`)
2) if `shouldCoerceStrings` is true, it attempts to coerce the String value to a bool, int or double

`convertAttributes` wasn't written to support lists so this is a bit of a hack